### DISCUSSION TITLE
mutate useSWRInfinity as global mutate

### DISCUSSION THREAD

**Discussion by ParkDyel** [vercel/swr#2880] [Q&A]
I am a developer who uses (or abuses) swr too comfortably. First of all, I would like to say thank you for this.
Please understand that I am not good at English, so I used Google Translator.

The gist of the question is this: I would like to use global mutate to update the data of useSwrInfinity even in places other than its children.

Data is updated well with the KeyedMutator bound to useSwrInfinity, but when the getKey function is passed to global mutate, the data is not immediately updated (network call) as intended.

Did I misunderstand the mutate or options of  useSWRInfinity?

Below is the code I tried.

```
  const getKey = (pageIndex: number) => {
    const start = pageIndex * perPage;

    if (start >= total) {
      return null; // reach end;
    }

    return `/v2/comment/${postId}?${stringify({
      page: pageIndex + 1,
      size: perPage,
    })}`;
  };

  const { data, error, size, setSize, isValidating, isLoading, mutate } =
    useSWRInfinite(
      getKey,
      url =>
        request<ICommunityResponse<ICommunityCommentItem>>({
          req: {
            method: 'GET',
            url: `${process.env.NEXT_PUBLIC_COMU_API_PREFIX}${url}`,
          },
        }).then(handleResponse(setTotal)),
      {
        revalidateIfStale: true,
        revalidateOnFocus: false,
        revalidateOnReconnect: false,
        revalidateAll: false,
        revalidateFirstPage: false,
        keepPreviousData: true,
        persistSize: false,
        initialSize: initialPage,
        parallel: true,
      },
    );
```

```
      mutate(
        key =>
          typeof key === 'string' && key.startsWith(`/v2/comment/${postId}`),
        undefined,
        { revalidate: true },
      );
```

I tried using the comparison operator directly in the key comparison function of mutate below, and the corresponding keys exist and the comparison statement returns true!

**Comment by koba04** [comment]
You are correct. Currently, `useSWRInfinite` doesn't work with the global `mutate`, so we are discussing the way to support the use case.
ref. https://github.com/vercel/swr/issues/2497
